### PR TITLE
Update sharing urls

### LINKED ISSUE #797: Some social sharing links do not work
It seems to me that some social sharing links that appear in the top right of the top bar do not work. I've checked it in properly hosted site. The URL of the website is **https://perso.in/testy/**

The links I checked:
1. Facebook - opens new window with error message. The link that I am redirected to is: https://www.facebook.com/sharer/sharer.php?s=100&p[url]=https%3A%2F%2Fperso.in%2Ftesty%2F
image
2. Twitter - opens main page but there is nothing inside tweet textbox. The link that I am redirected to is: https://twitter.com/home?status=Og%C3%B3lnopolskie%20Badanie%20Trener%C3%B3w%20-%20Raport%20https%3A%2F%2Fperso.in%2Ftesty%2F
3. LinkedIn - works just fine. I am redirected to the page where I can select wether I want to share it in private message or as a post. The link that I am redirected to is: https://www.linkedin.com/sharing/share-offsite/?url=https%3A%2F%2Fperso.in%2Ftesty%2F

Pandoc version 2.7.2. `xfun::session_info('bookdown')` output:
```
R version 3.6.1 (2019-07-05)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18362), RStudio 1.2.5001

Locale:
  LC_COLLATE=Polish_Poland.1250  LC_CTYPE=Polish_Poland.1250    LC_MONETARY=Polish_Poland.1250
  LC_NUMERIC=C                   LC_TIME=Polish_Poland.1250    

Package version:
  base64enc_0.1.3 bookdown_0.14.1 digest_0.6.22   evaluate_0.14   glue_1.3.1      graphics_3.6.1 
  grDevices_3.6.1 highr_0.8       htmltools_0.4.0 jsonlite_1.6    knitr_1.25      magrittr_1.5   
  markdown_1.1    methods_3.6.1   mime_0.7        Rcpp_1.0.2      rlang_0.4.1     rmarkdown_1.16 
  stats_3.6.1     stringi_1.4.3   stringr_1.4.0   tinytex_0.17    tools_3.6.1     utils_3.6.1    
  xfun_0.10       yaml_2.2.0   
```

---

By filing an issue to this repo, I promise that

- [x] I have fully read the issue guide at https://yihui.name/issue/.
- [x] I have provided the necessary information about my issue.
    - If I'm asking a question, I have already asked it on Stack Overflow or RStudio Community, waited for at least 24 hours, and included a link to my question there.
    - If I'm filing a bug report, I have included a minimal, self-contained, and reproducible example, and have also included `xfun::session_info('bookdown')`. I have upgraded all my packages to their latest versions (e.g., R, RStudio, and R packages), and also tried the development version: `remotes::install_github('rstudio/bookdown')`.
    - If I have posted the same issue elsewhere, I have also mentioned it in this issue.
- [x] I have learned the Github Markdown syntax, and formatted my issue correctly.

I understand that my issue may be closed if I don't fulfill my promises.

### PR DESCRIPTION
**Pull request by cderv**
This closes #797 

As explained in https://github.com/rstudio/bookdown/issues/797#issuecomment-549181261 there was changes in sharing url. 

This PR: 

* removes google+ as it does not exist anymore. 
* updates sharing link for twitter and facebook
* changes icon for instapaper as `fa-instapaper` does not exists.

`fa-italic`
image

Instapaper icone 
image

## About google removal

Google sharing is removed from example and default value. 
A new function `check_gb_config` is here to check configuration and change user's config with a warning to disable some option. (here remove google from `sharing` list or `all` vector

I think this is enough to not break older bookdown where `google` would have been present in configuration. 

We could also leave as is and the sharing link would redirect to google+ removal page
https://plus.google.com/

See for example https://bookdown.org/yihui/bookdown/ and sharing link to google plus.

### CODE REVIEW

File: R/gitbook.R [hunk 1]
```diff
@@ -284,3 +284,16 @@ download_filenames = function(config) {
   }
   if (length(downloads)) I(downloads)
 }
+
+check_gb_config = function(config) {
+  # to ensure backward compatibility with 0.14 and earlier
+  if (isTRUE(config[["sharing"]][["google"]]) ||
```
Comment (hadley) [inline R/gitbook.R]: This check breaks "`sharing: no`": https://travis-ci.org/hadley/mastering-shiny/builds/611371764#L721

Comment (cderv) [reply R/gitbook.R]: yes, I am really sorry about that. 
I worked on a fix as soon as I could yesterday evening and pushed a fix this morning with a simpler approach that should have less impacts. https://github.com/rstudio/bookdown/pull/813

Comment (yihui) [reply R/gitbook.R]: Should be fixed by `remotes::install_github('rstudio/bookdown')` now. Thanks!

Comment (GegznaV) [reply R/gitbook.R]: When can we expect these fixes on CRAN? As CRAN version of bookdown breaks the book building process if either `sharing: no` or `sharing: yes` are present.

Comment (yihui) [reply R/gitbook.R]: @GegznaV Unfortunately the fix came right after I released a new version of bookdown to CRAN, so I cannot update it again too soon. I wonder how difficult it is for you to install the development version.

Comment (GegznaV) [reply R/gitbook.R]: I was asking as I'm teaching a class of students. A part them downloaded the broken version of **bookdown**. I have to give recommendations on how to solve the issue and I usually prefer CRAN versions of packages. But as I see, in this situation, the GitHub version is the only option.

Comment (yihui) [reply R/gitbook.R]: I see. The earliest time I can make the next CRAN release is Nov 20 (I'll do it then). Thanks for your patience!

**Pull request merged**